cmd/boots/main.go

```diff
@@ -97,13 +138,163 @@ func main() {
 		}
 	}()
 
-	mainlog.Info("serving tftp")
-	go ServeTFTP()
-	mainlog.Info("serving dhcp")
-	go ServeDHCP()
+	var nextServer net.IP
+	var httpServerFQDN string
+	g, ctx := errgroup.WithContext(ctx)
+	ipxe := &ipxedust.Server{
+		Log:                  defaultLogger(cfg.logLevel),
+		EnableTFTPSinglePort: true,
+		TFTP:                 ipxedust.ServerSpec{Disabled: true},
+		HTTP:                 ipxedust.ServerSpec{Disabled: true},
+	}
+
+	if cfg.remoteTFTPAddr == "" { // use local iPXE binary service for TFTP
+		if !cfg.iTFTPDisabled {
+			ip := cfg.ipxe.TFTPAddr
+			if strings.Contains(ip, ":") {
+				ip = strings.Split(ip, ":")[0]
+				if port := strings.Split(cfg.ipxe.TFTPAddr, ":")[1]; port != "69" {
+					mainlog.With("providedPort", port).Info("warning: only port 69 is supported for TFTP")
+				}
+			}
+			ipportTFTP, err := netaddr.ParseIPPort(ip + ":69")
+			if err != nil {
+				mainlog.Fatal(err)
+			}
```

Comment:
follow up: I went looking through git history and saw the commit that made it a warning but there's no explanation for it. Please add more info to the commit message If you stick with the warning.

Reply:
https://github.com/tinkerbell/boots/pull/229/commits/b3807550c1fd2bc97b386da10d3bcee03ce22bae

_"github.com/packethost/dhcp4-go does not allow for specifying
a port for the TFTP server. This removes cli flags indicating
that a port should be specified. This also means that local or
remote TFTP servers must be listening on port 69 in order to
interoperate with Boots."_

This PR started out dropping support for a port in the local TFTP address, but the port was [requested](https://github.com/tinkerbell/boots/pull/229#issuecomment-1008262077) back in by @rgl. Because of the git comment above the port must be 69 and so we statically set it regardless of what the user provides. So the warning is just that. "only port 69 is supported for TFTP".

Reply:
In that message he also mentions that maybe we should panic if a different port was given. I don't think ignoring the user is the correct move here. We should either exit if an invalid port is given or go back to only taking an ip address and updating sandbox so it works there. The current setup just seems to be the most confusing route.

Reply:
Hey @rgl, curious about your thoughts here. Allowing a user the ability to specify a port feels kind of misleading. Like saying, "you can specify any port as long as it's port 69". Thoughts on only taking an IP for TFTP and then updating the sandbox? I do feel that since we can't allow customization of the port yet, we shouldn't allow the user to specify one.

Reply:
My only concern is about the cli argument names and their values; all but the tftp one receive a port:

```
  -dhcp-addr 192.168.121.96:67     IP and port to listen on for DHCP.
  -http-addr 192.168.121.96:80     local IP and port to listen on for the serving iPXE files via HTTP.
  -syslog-addr 192.168.121.96:514  IP and port to listen on for syslog messages.
  -tftp-addr 0.0.0.0               local IP to listen on for serving iPXE binaries via TFTP.
```

the `-addr` suffix seems to infer the same type/syntax can be used in all of them.

I'm fine in boots crashing when the user does not set a port to 69 (omitting it should be fine too; it would default to 69). 

Having a port seems more consistent and explicit.

But at the end of the day, that is just my opinion. I will use it anyways :-)

Reply:
@mmlb, @rgl  pushed changes to this. Thoughts?

Reply:
@jacobweinstock, looking good from here :-)